stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/ArtemisUtils.java

```diff
@@ -0,0 +1,159 @@
+            return getError(context, "Artemis connector error");
+        } else {
+            return getError(context, throwable.getMessage());
+        }
+    }
+
+    public static int getIntFromConfig(BMap<String, BValue> config, String key, Logger logger) {
+        return getIntFromLong(((BInteger) config.get(key)).intValue(), key, logger);
+    }
+
+    public static int getIntFromLong(long longVal, String key, Logger logger) {
+        if (longVal <= 0) {
+            return -1;
+        }
+        try {
+            return Math.toIntExact(longVal);
+        } catch (ArithmeticException e) {
+            logger.warn("The value set for {} needs to be less than {}. The {} value is set to {}", key,
+                        Integer.MAX_VALUE, key, Integer.MAX_VALUE);
+            return Integer.MAX_VALUE;
+        }
+    }
+
+    public static RoutingType getRoutingTypeFromConfig(String type) {
+        if ("MULTICAST".equals(type)) {
+            return RoutingType.MULTICAST;
+        }
+        return RoutingType.ANYCAST;
+    }
+
+    public static BValue getBValFromObj(Object obj, Context context) {
+        if (obj instanceof String) {
+            return new BString((String) obj);
+        } else if (obj instanceof Integer) {
+            return new BInteger((int) obj);
+        } else if (obj instanceof Long) {
+            return new BInteger((long) obj);
+        } else if (obj instanceof Short) {
+            return new BInteger((short) obj);
+        } else if (obj instanceof Float) {
+            return new BFloat((float) obj);
+        } else if (obj instanceof Double) {
+            return new BFloat((double) obj);
+        } else if (obj instanceof Boolean) {
+            return new BBoolean((boolean) obj);
+        } else if (obj instanceof Byte) {
+            return new BByte((byte) obj);
+        } else if (obj instanceof byte[]) {
+            return new BValueArray((byte[]) obj);
+        } else {
+            return ArtemisUtils.getError(context, new BallerinaException("Unsupported type"));
+        }
+    }
+
+    public static RoutingType getRoutingTypeFromString(String routingType) {
+        return ArtemisConstants.ANYCAST.equals(routingType) ? RoutingType.ANYCAST :
+                RoutingType.MULTICAST;
+    }
+
+    public static ClientSession getClientSessionFromBMap(BMap<String, BValue> obj) {
+        @SuppressWarnings(ArtemisConstants.UNCHECKED)
+        BMap<String, BValue> sessionObj = (BMap<String, BValue>) obj.get("session");
+        return (ClientSession) sessionObj.getNativeData(ArtemisConstants.ARTEMIS_SESSION);
+    }
+
+    public static void closeIfAnonymousSession(BMap<String, BValue> obj) throws ActiveMQException {
+        boolean anonymousSession = ((BBoolean) obj.get("anonymousSession")).booleanValue();
+        if (anonymousSession) {
+            ClientSession session = ArtemisUtils.getClientSessionFromBMap(obj);
+            if (session.isClosed()) {
```

Comment:
Why do you close again if the session is already closed? Shouldn't the condition be negation?